src/main/scala/com/snowflake/snowpark/functions.scala

```diff
@@ -3140,6 +3140,112 @@ object functions {
    */
   def listagg(col: Column): Column = listagg(col, "", isDistinct = false)
 
+  /**
+   * This function receives a column and extracts the groupIdx from the string
```

Comment:
lets try to align the description with: https://docs.snowflake.com/developer-guide/snowpark/reference/python/latest/snowpark/api/snowflake.snowpark.functions.regexp_extract and include code examples